src/Global/GlobalState.js

```diff
@@ -0,0 +1,140 @@
+import React, { useEffect, useState } from "react";
+import GlobalStateContext from "./GlobalStateContext"
+import axios from "axios";
+
+const GlobalState = (props) => {
+    //carrinho inicia
+    const [carrinho, setCarrinho] = useState([])
+    const [carrinhoRestaurantData, setCarrinhoRestaurantData] = useState({})
+    const [carrinhoDePostagem, setCarrinhoDePostagem] = useState([])
+    const [valorTotal, setValorTotal] = useState(0)
```

Comment:
Não há a necessidade do valor do carrinho ser um estado, pois ele pode ser computado com base no array de itens do carrinho no momento da renderização. Isso simplifica consideravelmente a lógica de atualização do carrinho.
Vejam os vídeos do Thinking In React, divulgados um tempão atrás com vocês, ou a aula em que resolvi o jogo dos 7 bugs com testes. Eu abordo exatamente esse problema nesses dois vídeos. Se quiserem e não conseguirem encontrar esses vídeos, podem mandar mensagem no Slack.